### ISSUE TITLE
On 0.4.0, "No Globalize compiled data module found" if app makes no direct use of Globalize

### ISSUE THREAD

**Issue by rxaviers** [rxaviers/react-globalize-webpack-plugin#14]
Impact:

On `0.4.0`, apps that uses react-globalize exclusively (i.e., without using Globalize directly) get `No Globalize compiled data module found` when compiling on production mode. The i18n chunks are build, but their order appears incorrectly in index.html, making app to fail loading.

Example: https://github.com/rxaviers/react-globalize-webpack-plugin/tree/example-for-14.

Quick workaround: Edit the built index.html and fix script loading order: vendor, i18n, app.

Cause:

On 0.4.x, when an app makes no direct use of `Globalize`, [`.createCompiledDataModule()`](https://github.com/rxaviers/globalize-webpack-plugin/blob/master/ProductionModePlugin.js#L77) is never called, therefore [no module in no chunks are created as a compiled module](https://github.com/rxaviers/globalize-webpack-plugin/blob/master/ProductionModePlugin.js#L127). Note that data from react-globalize are extracted and compiled just fine, it just happens they are not flagged/marked by globalize-webpack plugin. A fix might probably need to go in globalize-webpack-plugin, but I'm filing the issue here because it's related to react-globalize-webpack-plugin.

**Comment by rxaviers** [comment]
Demo for this issue https://github.com/rxaviers/react-globalize-webpack-plugin/tree/example-for-14. Steps: `npm install  && npm run build`.

**Comment by mehcode** [comment]
@rxaviers Anything I can do to help with this?

**Comment by rxaviers** [comment]
Hi @mehcode, it would be great if you could check the "Cause" in the description, use the demo above, and try to fix it. Just let me know if you have any questions. Thanks for your help.

**Comment by ambrons** [comment]
@rxaviers as you probably already know adding `import Globalize from 'globalize';` fixes the above mentioned issue.

[example fork](https://github.com/ambrons/react-globalize-webpack-plugin/commit/cd9752d36b1cc1c5a954d0816e065093dc8722b8)

**Comment by rxaviers** [comment]
Hey @ambrons, thanks for your observation. Although, (if I understood correctly) your suggestion requires a change in the user code. Let's find out a solution in the plugin itself.

**Comment by ambrons** [comment]
@rxaviers yes you're correct it's a user space fix.  Probably more of a _workaround_, but thought it was worth noting.  If I get some time I'll try to do a deeper dive.
